src/modules/services/cassandra.nix

```diff
@@ -0,0 +1,103 @@
+{ pkgs, lib, config, ... }:
+
+let
+  cfg = config.services.cassandra;
+  types = lib.types;
+  baseDir = config.env.DEVENV_STATE + "/cassandra";
+  JVM_OPTS = builtins.getEnv "JVM_OPTS";
+  cassandraConfig = lib.flip lib.recursiveUpdate cfg.extraConfig (
+    {
+      listen_address = cfg.listenAddress;
+      commitlog_sync = "batch";
+      commitlog_sync_batch_window_in_ms = 2;
+      cluster_name = cfg.clusterName;
+      partitioner = "org.apache.cassandra.dht.Murmur3Partitioner";
+      endpoint_snitch = "SimpleSnitch";
+      data_file_directories = [ "${baseDir}/data" ];
+      commitlog_directory = "${baseDir}/commitlog";
+      saved_caches_directory = "${baseDir}/saved_caches";
+      hints_directory = "${baseDir}/hints";
+      seed_provider = [
+        {
+          class_name = "org.apache.cassandra.locator.SimpleSeedProvider";
+          parameters = [{ seeds = lib.concatStringsSep "," cfg.seedAddresses; }];
+        }
+      ];
+    }
+  );
+  cassandraConfigFile = pkgs.writeText "cassandra.yaml" (builtins.toJSON cassandraConfig);
+  startScript = pkgs.writeShellScriptBin "start-cassandra" ''
+    set -euo pipefail
+
+    if [[ ! -d "${baseDir}" ]]; then
+      mkdir -p "${baseDir}"
+    fi
+
+    echo ${cassandraConfigFile}
+
+    exec ${cfg.package}/bin/cassandra -Dcassandra.config=file:///${cassandraConfigFile} -f
+  '';
+in
+{
+  imports = [
+    (lib.mkRenamedOptionModule [ "cassandra" "enable" ] [
+      "services"
+      "cassandra"
+      "enable"
+    ])
+  ];
+
+  options.services.cassandra = {
+    enable = lib.mkEnableOption "Add Cassandra process script.";
+
+    package = lib.mkOption {
+      type = types.package;
+      description = "Which version of Cassandra to use";
+      default = pkgs.cassandra;
+      defaultText = lib.literalExpression "pkgs.cassandra";
+      example = lib.literalExpression "pkgs.cassandra_3_0;";
+    };
+
+    listenAddress = lib.mkOption {
+      type = types.str;
+      description = "Listen address";
+      default = "127.0.0.1";
+      example = "127.0.0.1";
+    };
+
+    seedAddresses = lib.mkOption {
+      type = types.listOf types.str;
+      default = [ "127.0.0.1" ];
+      description = "The addresses of hosts designated as contact points of the cluster";
+    };
+
+    clusterName = lib.mkOption {
+      type = types.str;
+      default = "Test Cluster";
+      description = "The name of the cluster";
+    };
+
+    extraConfig = lib.mkOption {
+      type = types.attrs;
+      default = { };
+      example =
+        {
+          commitlog_sync_batch_window_in_ms = 3;
+        };
+      description = ''
+        Extra options to be merged into `cassandra.yaml` as nix attribute set.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    packages = [ cfg.package startScript ];
+
+    env.JVM_OPTS = (builtins.getEnv "JVM_OPTS") +
```

Comment:
I guess JVM_OPTS is a standard environment variable for Java apps. It would make sense to move this into `startScript` to it does not collidate with other java apps.

Then the `JVM_OPTS` could be also an option :)